lib/autobahn/consumer.rb

```diff
@@ -121,10 +120,17 @@ def setup!
     end
 
     def deliver(handler, timeout)
-      begin
+      while @deliver.get
         headers, message = self.next(timeout)
         handler.call(headers, message) if headers
-      end while @deliver.get
+      end
+      while (headers, message = self.next(0))
+        if headers
+          handler.call(headers, message)
+        else
+          break
+        end
+      end
```

Comment:
Not from what I can tell. The only difference would that we wouldn't call `#setup!` all of the time :)